pkg/parse/parse.go

```diff
@@ -387,12 +387,24 @@ func GetBindMount(args []string) (specs.Mount, error) {
 		}
 	}
 
+	// buildkit parity: default bind option must be `rbind`
+	// unless specified
+	if !bindNonRecursive {
+		newMount.Options = append(newMount.Options, "rbind")
+	}
+
 	if !setDest {
 		return newMount, noDestError
 	}
 
-	if !setSource {
-		newMount.Source = newMount.Destination
+	// buildkit parity: support absolute path for sources from current build context
+	if newMount.Source == "" || newMount.Source == "." || newMount.Source == "./" {
+		newMount.Source = contextDir
+	}
+	// if path begins with `./` replace `./` with context dir
+	if strings.HasPrefix(newMount.Source, "./") {
+		// path should be /contextDir/specified path
+		newMount.Source = filepath.Join(contextDir, filepath.Clean(string(filepath.Separator)+newMount.Source))
```

Comment:
Is this first block still needed?

Reply:
What's the intent here, specifically for source locations that aren't absolute paths?

Reply:
yes. builkit accepts paths starting with `.` and treats `.` as current context. However users are free to add any valid path from host machine.

Reply:
I would expect relative paths to be interpreted as relative to a context directory if a context directory was specified, and to the current directory otherwise.

Reply:
@nalind `ContextDir` falls back to `os.Getwd()` if nothing special is configured so yes this should happen.

Reply:
In `buildah build`, yes, but not in `buildah run` so far as I can see.

Reply:
This still needs to be something the tests cover.

Reply:
In particular, I'd like tests to cover cases where a source is specified with a location that looks like "tests/testdata" (i.e., with no leading "./" or "../", and not specified as an absolute path) when the current working directory when `buildah` is invoked is not the same as the context directory.  In every case, we need to be sure that we can't end up mounting a source location that is outside of the context directory.

Reply:
@nalind Yes i agree relative path should always be picked from `ContextDir` added a test could you PTAL